tests/Unit/Parallel/Test_PhaseChangeMain.cpp

```diff
@@ -172,13 +173,18 @@ struct ArrayComponent {
   static void allocate_array(
       Parallel::CProxy_GlobalCache<Metavariables>& global_cache,
       const tuples::tagged_tuple_from_typelist<initialization_tags>&
-      /*initialization_items*/) {
+      /*initialization_items*/,
+      const std::unordered_set<size_t>& procs_to_ignore = {}) {
     auto& local_cache = *Parallel::local_branch(global_cache);
     auto& array_proxy =
         Parallel::get_parallel_component<ArrayComponent>(local_cache);
 
-    for (int i = 0, which_proc = 0, number_of_procs = sys::number_of_procs();
-         i < 2; ++i) {
+    const size_t number_of_procs = static_cast<size_t>(sys::number_of_procs());
```

Comment:
can we factor this round robin code out somewhere? It's getting increasing complicated and so not having to update in a bunch of places would be nice